kaal-infrastructure/src/main/kotlin/cz/eman/kaal/infrastructure/api/retrofit/KaalRetrofitCaller.kt

```diff
@@ -176,13 +184,19 @@ open class KaalRetrofitCaller {
      * Handles call exception. Any exception that is handled is used to create [Result.Error]. It makes sure the call
      * does not crash the app and it receives information about what happened.
      *
+     * Override [rethrowExceptions] to specify the list of exceptions to re-throw.
+     *
      * @param ex Exception to handle
      * @param errorMessage used to modify [Result.Error] message
      * @return [Result] with type [T] - always creates [Result.Error]
      * @see errorResult
      * @since 0.9.0
      */
     private fun <T> handleCallException(ex: Exception, errorMessage: String?): Result<T> {
+        if (rethrowExceptions.isNotEmpty() && ex::class in rethrowExceptions) {
```

Comment:
Could it be only `if (ex::class in rethrowExceptions)` or not?

Reply:
Yea, I guess it is an unnecessary optimalization to check for non-empty.